### PR TITLE
eMIS code deletion for veteran_status

### PR DESCRIPTION
**Pull request by rmtolmach**
## Summary

- *This work is behind a feature toggle (flipper): NO*

This PR is a bite out of the eMIS code deletion for vets-api. This PR removes the old eMIS veteran status code (there is new VA Profile veteran status code being used in its place).

## Related issue(s)

- Part 1 from: https://github.com/department-of-veterans-affairs/va.gov-team/issues/72200
- Epic: https://github.com/department-of-veterans-affairs/va.gov-team/issues/63326

## Testing done
I used test user 99 because Rachal had success with that user when she was testing veteran_status for VA Profile. So this test is to make sure there were no regressions that might have been missed in unit tests. After logging into staging.va.gov as user 99, I opened up a terminal in argo staging and ran the following commands.
```
irb(main):001:0> UserAccount.find_by(icn:"<user 99's icn>").user_verifications.where.not(idme_uuid: nil).last.idme_uuid
=> "<icn_uuid>"
irb(main):002:0> 
irb(main):003:0> 
irb(main):004:0> 
irb(main):005:0> a = User.find("<icn_uuid>")
=> 
#<User:0x00007fa1a5971ca8
...
irb(main):006:0> a.can_access_id_card?
=> true
irb(main):007:0> 
irb(main):007:0> a.veteran_status
=> 
#<VAProfileRedis::VeteranStatus:0x00007fa1bbe3a210
 @persisted=false,
 @response=
  #<VAProfile::VeteranStatus::VeteranStatusResponse:0x00007fa1bbe36728
   @errors_hash={},
   @metadata={},
   @original_attributes=
    {:status=>nil,
     :title38_status_code=>
      #<VAProfile::Models::VeteranStatus:0x00007fa1bbe36570
       @title38_status_code="V1">},
   @status=200,
   @title38_status_code=
    #<VAProfile::Models::VeteranStatus:0x00007fa1bbe366b0
     @title38_status_code="V1">>,
...
```

## What areas of the site does it impact?
Since mid-December, eMIS has reported **no** traffic coming from **any** sources, so this should not impact anything (unless I've done something wrong! 😃) 

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.

### CODE REVIEW

**Review by jadon1979 — APPROVED** [review]

**Review by rjohnson2011 — APPROVED** [review]
Look great! Farewell eMIS!

File: app/services/users/exception_handler.rb [hunk 1]
```diff
@@ -11,15 +11,15 @@ class ExceptionHandler
...
+    # @param service [String] The name of the external service (i.e. 'Vet360', 'MVI', 'VA Profile')
     #
     def initialize(error, service)
       @error = validate!(error)
       @service = service
     end
 
     # Serializes the initialized error into one of the predetermined error types.
-    # Uses error classes that can be triggered by MVI, EMIS, or Vet360.
+    # Uses error classes that can be triggered by MVI, VA Profile, or Vet360 (aka VA Profile).
```
Comment (Mitch-A6) [inline app/services/users/exception_handler.rb:22]: 🤔 I may be missing context here, but this reads effectively like "can be triggered by... VA Profile... or... VA Profile" to me? Took me a moment to check back, thinking "wait, isn't that what it just said before? ... yes it is, huh".

Would something like
```
# Uses error classes that can be triggered by MVI, or VA Profile (aka Vet360).
```
carry the same meaning, or is there something else going on here?

Comment (rmtolmach) [reply app/services/users/exception_handler.rb]: Yeah, good point @Mitch-A6. Changed. Thanks!

**Review by ericboehs — APPROVED** [review]
Screenshot 2024-01-31 at 13 23 37@2x
jack-nicholson-yes-gif

**Review by rileyanderson — APPROVED** [review]

**Comment by rmtolmach** [comment]
Thanks for the reviews, everyone!

**Pull request merged**